xmake/modules/core/tools/gcc.lua

```diff
@@ -111,7 +111,7 @@ function nf_warning(self, level)
         none       = "-w"
     ,   less       = "-Wall"
     ,   more       = "-Wall"
-    ,   all        = "-Wall"
+    ,   all        = "-Wall -Wextra"
```

Comment:
Currently `all` warning level corresponds to -Wall, if we add `-Wextra`, it will cause too many warnings. This will affect all existing projects with `set_warnings("all", "error")` set, causing compilation interruption.

Reply:
We should add `-Wextra` to ʻeverything` warning level.

Reply:
-Wextra doesn't add too much warning in my opinion, here's the list:
```


-Wclobbered  
-Wcast-function-type  
-Wdeprecated-copy (C++ only) 
-Wempty-body  
-Wenum-conversion (C only) 
-Wignored-qualifiers 
-Wimplicit-fallthrough=3 
-Wmissing-field-initializers  
-Wmissing-parameter-type (C only)  
-Wold-style-declaration (C only)  
-Woverride-init  
-Wsign-compare (C only) 
-Wstring-compare 
-Wredundant-move (only for C++)  
-Wtype-limits  
-Wuninitialized  
-Wshift-negative-value (in C++03 and in C99 and newer)  
-Wunused-parameter (only with -Wunused or -Wall) 
-Wunused-but-set-parameter (only with -Wunused or -Wall)
```
(from https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html )

everything truly means "every possible warning" in my opinion, which Clangs supports with -Weverything, which is truly too much.

Reply:
Also, "more" already corresponds to -Wall, it's a bit odd to have "more" and "all" warning levels causing the same flags.

Reply:
> Also, "more" already corresponds to -Wall, it's a bit odd to have "more" and "all" warning levels causing the same flags.

Therefore, we should improve `more` to set warnings less than `-Wall`

Reply:
That's a possibility, but it means xmake is responsible of choosing which warnings are enabled by default on GCC/Clang, since they don't have other warnings groups than Wall and Wextra afaik.

Reply:
It is for this reason, currently I just correspond `more` to `-Wall` for gcc/clang.